src/lib/widgets/audio/ui/audio.svelte

```diff
@@ -0,0 +1,42 @@
+<script lang="ts">
+  import { audioStore } from '$lib/features/fetch-audio';
+  import ScrubbingBar from './scrubbing-bar.svelte';
+
+  let audioEl = null as HTMLAudioElement | null;
+  
+  let duration = 0;
+  let currentTime = 0;
+  let bufferTime = 0;
+  
+  const timeUpdate = () => {
+    if (!audioEl) return;
+    duration = audioEl.duration;
+    currentTime = audioEl.currentTime || 0;
+    if (audioEl.buffered.length) {
+      bufferTime = audioEl.buffered.end(audioEl.buffered.length - 1) || 0;
+    }
+  };
+  
+  audioStore.subscribe((val) => {
+    if (!audioEl) return;
+    audioEl.pause();
+    console.log('here', val);
+    audioEl.src = val;
+    audioEl.currentTime = 0;
+    audioEl.addEventListener('timeupdate', timeUpdate);
```

Comment:
This feels a little error prone in the future. I understand that attaching the same event listener multiple times will discard the extra instances, but if the listener got changed to something like `() => timeUpdate(time)` or anything similar, then it will be called many times.
Also, why is it not just `on:timeupdate={timeUpdate}` on the element itself?

Reply:
Thanks, your solution is neat